nucypher/blockchain/eth/chains.py

```diff
@@ -56,39 +67,122 @@ def __init__(self, interface: Union[BlockchainInterface, BlockchainDeployerInter
 
     def __repr__(self):
         class_name = self.__class__.__name__
-        r = "{}(interface={})"
-        return r.format(class_name, self.__interface)
+        r = "{}(interface={}, process={})"
+        return r.format(class_name, self.__interface, self.__provider_process)
 
     @property
     def interface(self) -> Union[BlockchainInterface, BlockchainDeployerInterface]:
         return self.__interface
 
+    @property
+    def peers(self):
+        if self._instance is NO_BLOCKCHAIN_AVAILABLE:
+            raise self.ConnectionNotEstablished
+        return self.interface.w3.geth.admin.peers()
+
+    @property
+    def syncing(self):
+        if self._instance is NO_BLOCKCHAIN_AVAILABLE:
+            raise self.ConnectionNotEstablished
+        return self.interface.w3.eth.syncing
+
+    def sync(self, timeout: int = 600):
+        """
+        Blocking call that polls the ethereum client for at least one ethereum peer
+        and knowledge of all blocks known by bootnodes.
+        """
+
+        # Record start time for timeout calculation
+        now = maya.now()
+        start_time = now
+
+        def check_for_timeout(timeout=timeout):
+            last_update = maya.now()
```

Comment:
weird naming thing - `last_update` just feels like `current_time`

Reply:
Were are we at all?!